File: ms_interference_manuscript.py
import numpy as np
import pandas as pd
import pingouin as pg
import scikit_posthocs as sp
import matplotlib.pyplot as plt
import scipy

########################################################################################################################
# Summary Stats

df = pd.read_csv('E:/Data/MSIT_MIND/merged_demo_behavioral_129.csv')

########################################################################################################################
# Compare RT in a mixed model ANOVA
df = pd.read_csv('E:/Data/MSIT_MIND/group_RT_129_rmanova.csv')

aovrm = pg.mixed_anova(dv='Response_Time', within='Condition', between='Group', subject='MIND_ID', data=df)
print(aovrm.round(3))
pg.print_table(aovrm)

df_con = df[df['Group'] == 'Control']
ms = df_con[df['Condition'] == 'MultiSource']['Response_Time'].values
flanker = df_con[df['Condition'] == 'Flanker']['Response_Time'].values
simon = df_con[df['Condition'] == 'Simon']['Response_Time'].values
control = df_con[df['Condition'] == 'Control']['Response_Time'].values

ms_con = scipy.stats.ttest_rel(ms, control)
ms_flanker = scipy.stats.ttest_rel(ms, flanker)
ms_simon = scipy.stats.ttest_rel(ms, simon)

flanker_con = scipy.stats.ttest_rel(flanker, control)
flanker_simon = scipy.stats.ttest_rel(flanker, simon)

simon_con = scipy.stats.ttest_rel(simon, control)

df_hiv = df[df['Group'] == 'HIV']
ms = df_hiv[df['Condition'] == 'MultiSource']['Response_Time'].values
flanker = df_hiv[df['Condition'] == 'Flanker']['Response_Time'].values
simon = df_hiv[df['Condition'] == 'Simon']['Response_Time'].values
control = df_hiv[df['Condition'] == 'Control']['Response_Time'].values

ms_con = scipy.stats.ttest_rel(ms, control)
ms_flanker = scipy.stats.ttest_rel(ms, flanker)
ms_simon = scipy.stats.ttest_rel(ms, simon)

flanker_con = scipy.stats.ttest_rel(flanker, control)
flanker_simon = scipy.stats.ttest_rel(flanker, simon)

simon_con = scipy.stats.ttest_rel(simon, control)

########################################################################################################################
# Compare Accuracy in a Friedman's test

df = pd.read_csv('E:/Data/MSIT_MIND/accuracy_129_friedman.csv')

df_con = df[df['Group'] == 'Control']
ftest = scipy.stats.friedmanchisquare(df_con[df_con['Condition'] == 'Control']['Accuracy'].values,
                              df_con[df_con['Condition'] == 'Simon']['Accuracy'].values,
                              df_con[df_con['Condition'] == 'Flanker']['Accuracy'].values,
                              df_con[df_con['Condition'] == 'MultiSource']['Accuracy'].values)

posthoc = sp.posthoc_conover(df_con, val_col='Accuracy', group_col='Condition', p_adjust='bonf')

df_hiv = df[df['Group'] == 'HIV']
ftest = scipy.stats.friedmanchisquare(df_hiv[df_hiv['Condition'] == 'Control']['Accuracy'].values,
                              df_hiv[df_hiv['Condition'] == 'Simon']['Accuracy'].values,
                              df_hiv[df_hiv['Condition'] == 'Flanker']['Accuracy'].values,
                              df_hiv[df_hiv['Condition'] == 'MultiSource']['Accuracy'].values)

posthoc = sp.posthoc_conover(df_hiv, val_col='Accuracy', group_col='Condition', p_adjust='bonf')

mann_whit = scipy.stats.mannwhitneyu(df_con['Accuracy'].values, df_hiv['Accuracy'].values)

########################################################################################################################
# Potential superadditive effects in RT

df = pd.read_csv('E:/Data/MSIT_MIND/group_RT_129_modified_interference.csv')
aovrm = pg.mixed_anova(dv='Response_Time', within='Condition', between='Group', subject='MIND_ID', data=df)
pg.print_table(aovrm)

add = df[df['Condition'] == 'Additive']['Response_Time'].values
ms = df[df['Condition'] == 'MultiSource']['Response_Time'].values

p_ttest = scipy.stats.ttest_rel(add, ms)

# Potential Superadditive in Accuracy

df = pd.read_csv('E:/Data/MSIT_MIND/accuracy_129_additive.csv')

df_con = df[df['Group'] == 'Control']
add = df_con[df_con['Condition'] == 'Additive']['Accuracy_Effect'].values
ms = df_con[df_con['Condition'] == 'Multisource']['Accuracy_Effect'].values

wilc = scipy.stats.wilcoxon(add, ms)

df_hiv = df[df['Group'] == 'HIV']
add = df_hiv[df_hiv['Condition'] == 'Additive']['Accuracy_Effect'].values
ms = df_hiv[df_hiv['Condition'] == 'Multisource']['Accuracy_Effect'].values

wilc = scipy.stats.wilcoxon(add, ms)

mann_whit = scipy.stats.mannwhitneyu(df_con['Accuracy_Effect'].values, df_hiv['Accuracy_Effect'].values)


########################################################################################################################
# Alpha Interaction

df = pd.read_csv('E:/Data/MSIT_MIND/VMPs/Visual_4mm/Extracted_Peaks/additive_alpha_50_20_-3.csv')

aovrm = pg.mixed_anova(dv='pseudo_tvalue', within='Condition', between='Group', subject='ID', data=df)
print(aovrm.round(3))
pg.print_table(aovrm)

df_control = df[df['Group'] == 'Control']
df_hiv = df[df['Group'] == 'HIV']

aovrm = pg.rm_anova(dv='pseudo_tvalue', within='Condition', subject='ID', data=df_control)
pg.print_table(aovrm)

aovrm = pg.rm_anova(dv='pseudo_tvalue', within='Condition', subject='ID', data=df_hiv)
pg.print_table(aovrm)

ttest = pg.pairwise_ttests(dv='pseudo_tvalue', within='Condition', between='Group', subject='ID', padjust='bonf', data=df)
print(ttest[['Contrast', 'Condition', 'A', 'B', 'T', 'p-corr']])

########################################################################################################################
# Plot for Correlation Between Neural and Behavioral Interference Effect

import seaborn as sns

df = pd.read_csv('E:/Data/MSIT_MIND/VMPs/Visual_4mm/MS_interference_correlation/alpha/alpha_-7_29_1.csv')

plt.figure(figsize=(10, 10))
p = sns.lmplot(x='pseudo_tvalue', y='MS_Int', hue='Group', data=df, height=5,
                 palette=dict(Control='#F8766D', HIV='#00BFC4'), truncate=False, ci=False)
p.set(xlabel='Pseudo-t value', ylabel='Multisource Interference (ms)')
plt.savefig('E:/Data/MSIT_MIND/Manuscript/v1/Figures/alpha_-7_29_1_correlation.png', dpi=500)

df = pd.read_csv('E:/Data/MSIT_MIND/VMPs/Visual_4mm/MS_interference_correlation/alpha/alpha_-14_12_37.csv')

plt.figure(figsize=(10, 10))
p = sns.lmplot(x='pseudo_tvalue', y='MS_Int', hue='Group', data=df, height=5,
                 palette=dict(Control='#F8766D', HIV='#00BFC4'), truncate=False, ci=False)
p.set(xlabel='Pseudo-t value', ylabel='Multisource Interference (ms)')
plt.savefig('E:/Data/MSIT_MIND/Manuscript/v1/Figures/alpha_-14_12_37_correlation.png', dpi=500)

df = pd.read_csv('E:/Data/MSIT_MIND/VMPs/Visual_4mm/MS_interference_correlation/gamma/gamma_-44_9_41.csv')

plt.figure(figsize=(10, 10))
p = sns.lmplot(x='pseudo_tvalue', y='MS_Int', hue='Group', data=df, height=5,
                 palette=dict(Control='#F8766D', HIV='#00BFC4'), truncate=False, ci=False)
p.set(xlabel='Pseudo-t value', ylabel='Multisource Interference (ms)')
plt.savefig('E:/Data/MSIT_MIND/Manuscript/v1/Figures/gamma_-44_9_41_correlation.png', dpi=500)

df = pd.read_csv('E:/Data/MSIT_MIND/VMPs/Visual_4mm/MS_interference_correlation/gamma/gamma_17_-62_23.csv')

plt.figure(figsize=(10, 10))
p = sns.lmplot(x='pseudo_tvalue', y='MS_Int', hue='Group', data=df, height=5,
                 palette=dict(Control='#F8766D', HIV='#00BFC4'), truncate=False, ci=False)
p.set(xlabel='Pseudo-t value', ylabel='Multisource Interference (ms)')
plt.savefig('E:/Data/MSIT_MIND/Manuscript/v1/Figures/gamma_17_-62_23_correlation.png', dpi=500)

File: ms_interference_manuscript.R
library(ggplot2)

###############################################################
# RT plot
# Reorganize RT data for plot
rt_df = read.csv('E:/Data/MSIT_MIND/group_RT_129_modified_v2.csv')

rt_df = rt_df[which(rt_df$Condition == 'Control' | rt_df$Condition == 'Simon' | rt_df$Condition == 'Flanker' | rt_df$Condition == 'MultiSource'),]

rt_df$Condition <- factor(rt_df$Condition, 
                          levels = c('Control', 'Simon', 'Flanker', 'MultiSource'),
                          labels = c('Congruent', 'Simon', 'Flanker', 'MultiSource'))
tiff('E:/Data/MSIT_MIND/Manuscript/v1/Figures/response_times.png', units='in', width=5, height=5, res=300)
# ggboxplot(rt_df, x='Group', y='Response_Time', color='Group', ylab='Response Time (ms)')
# ggbarplot(rt_df, x='Condition', y='Response_Time', fill='Group', color='Group',
#           ylab='Response Time (ms)', position = position_dodge(.9), add = 'mean_se', ylim=c(500,900))
ggboxplot(rt_df, x='Condition', y='Response_Time', color='Group', ylab='Response Time (ms)')
dev.off()


###############################################################
# Response Time Shapiro Wilk Test for Normality
rt_df = read.csv('E:/Data/MSIT_MIND/group_RT_129_modified_v2.csv')

rt_cont = rt_df[which(rt_df$Condition == 'Control'),]
rt_simon = rt_df[which(rt_df$Condition == 'Simon'),]
rt_flan = rt_df[which(rt_df$Condition == 'Flanker'),]
rt_ms = rt_df[which(rt_df$Condition == 'MultiSource'),]

# Response Time Normality by Group and Condition

shapiro.test(rt_cont$Response_Time)
shapiro.test(rt_simon$Response_Time)
shapiro.test(rt_flan$Response_Time)
shapiro.test(rt_ms$Response_Time)

###############################################################
# Accuracy Shapiro Wilk Test for Normality

df = read.csv('E:/Data/MSIT_MIND/accuracy_129_friedman.csv')

shapiro.test(df$Control_Accuracy)
shapiro.test(df$Simon_Accuracy)
shapiro.test(df$Flanker_Accuracy)
shapiro.test(df$MultiSource_Accuracy)

# Accuracy Plot

df = read.csv('E:/Data/MSIT_MIND/accuracy_129_friedman.csv')

df = df[which(df$Condition == 'Control' | df$Condition == 'Simon' | df$Condition == 'Flanker' | df$Condition == 'MultiSource'),]

df$Accuracy = df$Accuracy *100

df$Condition <- factor(df$Condition, 
                          levels = c('Control', 'Simon', 'Flanker', 'MultiSource'),
                          labels = c('Congruent', 'Simon', 'Flanker', 'MultiSource'))
# ggbarplot(df, x='Condition', y='Accuracy', fill='Group', color='Group',
          # ylab='Accuracy (%)', position = position_dodge(.9), add = 'mean_se', ylim = c(.8, 1))

tiff('E:/Data/MSIT_MIND/Manuscript/v1/Figures/accuracy.png', units='in', width=5, height=5, res=300)
ggboxplot(df, x='Condition', y='Accuracy', color='Group', ylab='Accuracy (% correct)')
dev.off()


###############################################################
# Additive vs. Multisource Figure

# Additive vs. MultiSource Alpha 2x2 Interaction

df = read.csv('E:/Data/MSIT_MIND/VMPs/Visual_4mm/Extracted_Peaks/additive_alpha_50_20_-3.csv')
df$Condition <- factor(df$Condition, 
                       levels = c('Additive', 'MultiSource'),
                       labels = c('Additive', 'MultiSource'))
ggbarplot(df, x='Condition', y='pseudo_tvalue', fill='Group', color='Group', position = position_dodge(.9), add='mean_sd')

ggboxplot(df, x='Condition', y='pseudo_tvalue')

ggplot(df, aes(x=pseudo_tvalue, color=Condition)) + 
  geom_histogram(binwidth=1)

###############################################################
# Misc

df = read.csv('E:/Data/MSIT_MIND/accuracy_129_additive_fixed.csv')

df$Condition <- factor(df$Condition, 
                       levels = c('Additive', 'Multisource'),
                       labels = c('Additive', 'Multisource'))

ggboxplot(df, x='Condition', y='Accuracy_int_per', add='jitter', shape='Condition') + scale_y_reverse()





